js/StickyScrollBarComponent.js

```diff
@@ -0,0 +1,138 @@
+        this.$elementWithStickyScrollBar = {};
+        this.$window = $rootWindow;
+        this.$html = $html;
+        this.$stickyScrollBar = $('<div class="sticky-scrollbar"><div class="sticky-scrollbar__inner"></div></div>');
+        this.$stickyScrollBarInner = this.$stickyScrollBar.children();
+        // cache methods with context
+        this.scrollStickyScrollBarWithContext = this.scrollStickyScrollBar.bind(this);
+        this.scrollElementWithStickyScrollBarByScrollBarWithContext = this.scrollElementWithStickyScrollBarByScrollBar.bind(this);
+        this.updateStickyScrollBarWithContext = this.updateStickyScrollBar.bind(this);
+    }
+
+    /**
+     * Initialise
+     * @param {jQuery} $element - jQuery object of the element that requires a sticky scroll bar
+     */
+    init ($element) {
+        if (typeof $element === 'undefined' || !$element) {
+            throw new Error('an element to add the scroller to must be passed to StickyScrollBarComponent');
+        }
+
+        if (typeof this.$window === 'undefined' || !this.$window) {
+            throw new Error('window must be passed to StickyScrollBarComponent');
+        }
+
+        if (typeof this.$html === 'undefined' || !this.$html) {
+            throw new Error('$html must be passed to StickyScrollBarComponent');
+        }
+
+        // Set $elementWithStickyScrollBar
+        this.$elementWithStickyScrollBar = $element;
+    
+        // Add the sticky scroll bar
+        this.$stickyScrollBar.appendTo(this.$elementWithStickyScrollBar);
+
+        // Init the sticky scroll bar
+        this.$stickyScrollBar
+            .addClass('u-display-none')
+            .on('scroll', this.scrollElementWithStickyScrollBarByScrollBarWithContext);
+
+        // Call on load
+        this.updateStickyScrollBarWithContext();
+
+        // Call on scroll and window resize
+        this.$window.on('scroll resize', this.updateStickyScrollBarWithContext);
+    }
+
+    /**
+     * Set $elementWithStickyScrollBar scroll when scrolled by sticky scroll bar
+     */
+    scrollElementWithStickyScrollBarByScrollBar () {
+        this.$elementWithStickyScrollBar.scrollLeft(this.$stickyScrollBar.scrollLeft());
+    }
+
+    /**
+     * Set $stickyScrollBar scroll when $elementWithStickyScrollBar is scrolled
+     */
+    scrollStickyScrollBar () {
+        this.$stickyScrollBar.scrollLeft(this.$elementWithStickyScrollBar.scrollLeft());
+    }
+
+    /**
+     * Toggle visiblity of sticky scroll bar
+     */
+    showStickyScrollBar (option) {
+        if (option) {
+            this.$stickyScrollBar.removeClass('u-display-none');
+        } else {
+            this.$stickyScrollBar.addClass('u-display-none');
+        }
+    }
+
+    /**
+     * Update sticky scroll bar visibility, thumb position and width
+     */
+    updateStickyScrollBar () {
+        const top = this.$elementWithStickyScrollBar.offset().top;
+        const bottom = top + this.$elementWithStickyScrollBar.height();
```

Comment:
Really minor so ignore, but in the pursuit of locking the FPS as high as we can, we could reduce the browser "layout" or sometimes referred to as "reflow" by grouping these rect queries. jQuery will make 2 calls here to `element.getBoundingClientRect(...)` to get your `offset.top` and `height`, if we simplified this we could make 1 call to get both values...
```
const { top, height } = this.$elementWithStickyScrollBar[0].getBoundingClientRect();
```

I'm not suggesting this is worth changing as [premature optimisation is the root of all evil](https://cdb.reacttraining.com/react-inline-functions-and-performance-bdff784f5578), but something I like to keep at the back of my mind, in terms of expensive DOM operations.